File: src/Http.API/ClientApp/src/app/share/services/system-role.service.ts
import { Injectable } from '@angular/core';
import { BaseService } from './base.service';
import { Observable } from 'rxjs';
import { SystemRoleFilterDto } from '../models/system-role/system-role-filter-dto.model';
import { SystemRoleAddDto } from '../models/system-role/system-role-add-dto.model';
import { SystemRoleUpdateDto } from '../models/system-role/system-role-update-dto.model';
import { SystemRoleItemDtoPageList } from '../models/system-role/system-role-item-dto-page-list.model';
import { SystemRole } from '../models/system-role/system-role.model';

/**
 * SystemRole
 */
@Injectable({ providedIn: 'root' })
export class SystemRoleService extends BaseService {
  /**
   * filter
   * @param data SystemRoleFilterDto
   */
  filter(data: SystemRoleFilterDto): Observable<SystemRoleItemDtoPageList> {
    const url = `/api/SystemRole/filter`;
    return this.request<SystemRoleItemDtoPageList>('post', url, data);
  }

  /**
   * add
   * @param data SystemRoleAddDto
   */
  add(data: SystemRoleAddDto): Observable<SystemRole> {
    const url = `/api/SystemRole`;
    return this.request<SystemRole>('post', url, data);
  }

  /**
   * update
   * @param id string
   * @param data SystemRoleUpdateDto
   */
  update(id: string, data: SystemRoleUpdateDto): Observable<SystemRole> {
    const url = `/api/SystemRole/${id}`;
    return this.request<SystemRole>('put', url, data);
  }

  /**
   * getDetail
   * @param id string
   */
  getDetail(id: string): Observable<SystemRole> {
    const url = `/api/SystemRole/${id}`;
    return this.request<SystemRole>('get', url);
  }

  /**
   * delete
   * @param id string
   */
  delete(id: string): Observable<SystemRole> {
    const url = `/api/SystemRole/${id}`;
    return this.request<SystemRole>('delete', url);
  }

}

File: src/Sdks/Dusi.Manage.Client/Services/SystemRoleService.cs
using Share.Models.SystemRoleDtos;
namespace Dusi.Manage.Client.Services;
/// <summary>
/// 角色表
/// </summary>
public class SystemRoleService : BaseService
{
    public SystemRoleService(HttpClient httpClient) : base(httpClient)
    {

    }
    /// <summary>
    /// 筛选
    /// </summary>
    /// <param name="data">SystemRoleFilterDto</param>
    /// <returns></returns>
    public async Task<PageList<SystemRoleItemDto>?> Filter(SystemRoleFilterDto data) {
        var url = $"/api/admin/SystemRole/filter";
        return await PostJsonAsync<PageList<SystemRoleItemDto>?>(url, data);
    }

    /// <summary>
    /// 新增
    /// </summary>
    /// <param name="data">SystemRoleAddDto</param>
    /// <returns></returns>
    public async Task<SystemRole?> Add(SystemRoleAddDto data) {
        var url = $"/api/admin/SystemRole";
        return await PostJsonAsync<SystemRole?>(url, data);
    }

    /// <summary>
    /// 更新
    /// </summary>
    /// <param name="id"> </param>
    /// <param name="data">SystemRoleUpdateDto</param>
    /// <returns></returns>
    public async Task<SystemRole?> Update(string id, SystemRoleUpdateDto data) {
        var url = $"/api/admin/SystemRole/{id}";
        return await PutJsonAsync<SystemRole?>(url, data);
    }

    /// <summary>
    /// 详情
    /// </summary>
    /// <param name="id"> </param>
    /// <returns></returns>
    public async Task<SystemRole?> GetDetail(string id) {
        var url = $"/api/admin/SystemRole/{id}";
        return await GetJsonAsync<SystemRole?>(url);
    }

    /// <summary>
    /// ⚠删除
    /// </summary>
    /// <param name="id"> </param>
    /// <returns></returns>
    public async Task<SystemRole?> Delete(string id) {
        var url = $"/api/admin/SystemRole/{id}";
        return await DeleteJsonAsync<SystemRole?>(url);
    }

}